app/controllers/teams_controller.rb

```diff
@@ -42,11 +42,9 @@ def destroy
 
   def add_user
     authorize team, :update?
-    if includes_user?
-      return redirect_to_edit_team(:error, t('teams.users.add.exists'))
-    end
-    team.users << User.find(team_params[:user_id])
-    redirect_to_edit_team(:success, t('teams.users.add.success'))
+    user = User.find_by_id(team_params[:user_id]) || User.find_by_email(team_params[:email])
+    return add_existent_user(user) if user.present?
+    add_new_invitable_user
```

Comment:
We could make use of DCI here. The current user is inviting the new user to join Codestats. I know there isn't much code to move away from the controller, but we can use it to polish the way in which we use DCI. What do you think?